api/base/permissions.py

```diff
@@ -53,6 +54,12 @@ def _get_scopes(self, request, view):
             except AttributeError:
                 raise ImproperlyConfigured('TokenHasScope requires the view to define the '
                                            'required_read_scopes attribute')
+            assert is_iterable_but_not_string(view.required_read_scopes), \
+                'The required_read_scopes must be an iterable of CoreScopes'
+            if view.required_read_scopes and isinstance(view.required_read_scopes[0], tuple):
```

Comment:
Longer than it takes to not run it, and with much the same effect.

Performance is the cumulative result of many small design decisions. This one won't make or break us by itself, but it's worth asking ourselves the question for every new feature we push.